### PR TITLE
add easyblock for OCaml with support for extensions + generic OCamlPackage easyblock

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by wpoely86** [comment]
This seems ready to merge no?

File: easybuild/easyblocks/o/ocaml.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+        if not 'world.opt' in self.cfg['buildopts']:
+            self.cfg.update('buildopts', 'world.opt')
+
+        super(EB_OCaml, self).configure_step()
+
+    def install_step(self):
+        """
+        Custom install procedure for OCaml.
+        First install OCaml using 'make install', then install OPAM (if sources are provided).
```
Comment (wpoely86) [inline easybuild/easyblocks/o/ocaml.py]: `OPAM` is the package manager?

Comment (boegel) [inline easybuild/easyblocks/o/ocaml.py]: yes, there's pretty much no other way to install OCaml packages (somewhat reliably)

**Comment by boegel** [comment]
Good to go, tested with https://github.com/hpcugent/easybuild-easyconfigs/pull/2016, thanks for the review @wpoely86!

**Pull request merged**